modules/endpoint/variables.tf

```diff
@@ -0,0 +1,8 @@
+##############################################################################
+# Common variables
+##############################################################################
+
+variable "region" {
+  description = "Region used to determine the IBM Cloud Container Registry endpoint"
```

Comment:
This should have variable validation because the module can only support what you have in the hard coded map. Perhaps also reference https://cloud.ibm.com/docs/Registry?topic=Registry-registry_overview#registry_regions for regions.